### ISSUE TITLE
“Create .resx files” is disabled at Export to Project 

### ISSUE THREAD

**Issue by escms** [0xd4d/dnSpy#1061]
![tim20190125120628](https://user-images.githubusercontent.com/7844745/51724776-d2c93e80-2099-11e9-8af8-95f88eec8575.jpg)

**Comment by 0xd4d** [comment]
It's been disabled until a new resource reader is used, one that doesn't deserialize random stuff from the binary.

**Comment by rolandh** [comment]
Any updates on this?

Random stuff is better than nothing in some cases. Maybe just put a warning next to it?

**Comment by Sonic-The-Hedgehog-LNK1123** [comment]
> 
> 
> It's been disabled until a new resource reader is used, one that doesn't deserialize random stuff from the binary.

I downloaded dnSpy 3.0.2 to test this on some files, and I don't see any "random serialization" at all.
@0xd4d  If you are talking about the base64 encoded data, I can tell you: **this is normal behavior**.

If your .resx file reference external binary files, this reference information is lost during compilation. So, when you convert .resources files back to .resx files, this data is base64 encoded in the .resx file itself instead.

@0xd4d  If the above describes your serialization "problem", then I can tell you: There is no problem, add this feature back ASAP.

**Comment by CreateAndInject** [comment]
@Sonic-The-Hedgehog-LNK1123 What 0xd4d say is .Net resources reader will load assembly into memory, if it contains virus, it's dangerous to your computer.

@0xd4d By the way, I leave a comment here:
https://github.com/0xd4d/de4dot/commit/5fdd91ae0593a8339f2f6a64e7e02d1e8d89d0d0
Could you tell whether my change is right and safe?

**Comment by 0xd4d** [comment]
Two options: use another program or hack the source code:

https://github.com/0xd4d/dnSpy/blob/master/dnSpy/dnSpy/Documents/Tabs/Dialogs/ExportToProjectVM.cs#L166

https://github.com/0xd4d/dnSpy/blob/master/dnSpy/dnSpy.Decompiler/MSBuild/ProjectModuleOptions.cs#L68


@CreateAndInject 
Create a repro to verify your theory then send a PR

**Comment by Sonic-The-Hedgehog-LNK1123** [comment]
@CreateAndInject Of course, I didn't really think about that while writing my comment. Thanks for the clarification.

In that case, I agree to @rolandh, Just give the user a warning that code from the selected assembly will be executed, and the option should only be used on assemblies the user fully trusts.

The warning from snaker's generic unpacker for PEiD reads:

> WARNING: The target is executed both during OEP detection and during unpacking. The author is not responsible for any damages caused due to the use of this program. Please don't try this with suspicious files.

IMO, giving the user a warning like that will be appropriate. 

BTW, good luck implementing a custom reader, It will be a challenge to deal with custom serializers.

**Comment by CreateAndInject** [comment]
@0xd4d I have attached a .exe to repro the issue about infinite loop, please check it:
https://github.com/0xd4d/de4dot/commit/5fdd91ae0593a8339f2f6a64e7e02d1e8d89d0d0
Could you tell me whether my change is right and safe?
